src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -942,14 +942,14 @@ public ImmutableArray<T> Sort(int index, int count, IComparer<T> comparer)
             Requires.Range(index >= 0, "index");
             Requires.Range(count >= 0 && index + count <= self.Length, "count");
 
-            if (comparer == null)
-            {
-                comparer = Comparer<T>.Default;
-            }
-
             // 0 and 1 element arrays don't need to be sorted.
             if (count > 1)
             {
+                if (comparer == null)
```

Comment:
This change to move the comparer check and setting is fine from my perspective, but it doesn't have anything to do with wrapping/unwrapping immutable arrays, right?